exporter/clickhousetracesexporter/writerV3.go

```diff
@@ -0,0 +1,431 @@
+					return err
+				}
+			}
+			// add mapOfSpanAttributeKey to map
+			mapOfSpanAttributeKeys[mapOfSpanAttributeKey] = struct{}{}
+
+			if spanAttribute.DataType == "string" {
+				err = tagStatement.Append(
+					time.Unix(0, int64(span.StartTimeUnixNano)),
+					spanAttribute.Key,
+					spanAttribute.TagType,
+					spanAttribute.DataType,
+					spanAttribute.StringValue,
+					nil,
+					spanAttribute.IsColumn,
+				)
+			} else if spanAttribute.DataType == "float64" {
+				err = tagStatement.Append(
+					time.Unix(0, int64(span.StartTimeUnixNano)),
+					spanAttribute.Key,
+					spanAttribute.TagType,
+					spanAttribute.DataType,
+					nil,
+					spanAttribute.NumberValue,
+					spanAttribute.IsColumn,
+				)
+			} else if spanAttribute.DataType == "bool" {
+				err = tagStatement.Append(
+					time.Unix(0, int64(span.StartTimeUnixNano)),
+					spanAttribute.Key,
+					spanAttribute.TagType,
+					spanAttribute.DataType,
+					nil,
+					nil,
+					spanAttribute.IsColumn,
+				)
+			}
+			if err != nil {
+				w.logger.Error("Could not append span to tag Statement batch due to error: ", zap.Error(err), zap.Any("span", span))
+				return err
+			}
+		}
+	}
+
+	tagStart := time.Now()
+	err = tagStatement.Send()
+	stats.RecordWithTags(ctx,
+		[]tag.Mutator{
+			tag.Upsert(exporterKey, pipeline.SignalTraces.String()),
+			tag.Upsert(tableKey, w.attributeTable),
+		},
+		writeLatencyMillis.M(int64(time.Since(tagStart).Milliseconds())),
+	)
+	if err != nil {
+		w.logger.Error("Could not write to span attributes table due to error: ", zap.Error(err))
+		return err
+	}
+
+	tagKeyStart := time.Now()
+	err = tagKeyStatement.Send()
+	stats.RecordWithTags(ctx,
+		[]tag.Mutator{
+			tag.Upsert(exporterKey, pipeline.SignalTraces.String()),
+			tag.Upsert(tableKey, w.attributeKeyTable),
+		},
+		writeLatencyMillis.M(int64(time.Since(tagKeyStart).Milliseconds())),
+	)
+	if err != nil {
+		w.logger.Error("Could not write to span attributes key table due to error: ", zap.Error(err))
+		return err
+	}
+
+	return err
+}
+
+// WriteBatchOfSpans writes the encoded batch of spans
+func (w *SpanWriter) WriteBatchOfSpansV3(ctx context.Context, batch []*SpanV3, metrics map[string]usage.Metric) error {
+	var wg sync.WaitGroup
+	var chErr = make(chan error, 3)
+
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		err := w.writeIndexBatchV3(ctx, batch)
+		if err != nil {
+			w.logger.Error("Could not write a batch of spans to index table: ", zap.Error(err))
+			chErr <- err
+		}
+	}()
+
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		err := w.writeErrorBatchV3(ctx, batch)
+		if err != nil {
+			w.logger.Error("Could not write a batch of spans to error table: ", zap.Error(err))
+			chErr <- err
+		}
+	}()
+
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		err := w.writeTagBatchV3(ctx, batch)
+		if err != nil {
+			w.logger.Error("Could not write a batch of spans to tag/tagKey tables: ", zap.Error(err))
+			chErr <- err
+		}
```

Comment:
I wouldn't propagate the error from here since this is a metadata table. Failing the whole export because of the metadata table is not a good idea.